File: src/module/fortran/latc8m_struct.f
      MODULE latcm8m_struct
            USE array_structs
            INCLUDE '../../krcc8m.f'
            INCLUDE '../../latc8m.f'

            
            TYPE, BIND(C) :: LATCOM
                  integer(C_INT) :: NWLAT
                  type(REAL_ARRAY) :: DTM4
                  type(REAL_ARRAY) :: TST4
                  type(REAL_ARRAY) :: TTS4
                  type(REAL_ARRAY) :: TTB4
                  type(REAL_ARRAY) :: FROST4
                  type(REAL_ARRAY) :: AFRO4
                  type(REAL_ARRAY) :: TTA4
                  type(REAL_ARRAY) :: TTX4
                  type(REAL_ARRAY) :: TMN4
                  type(REAL_ARRAY) :: TIN
                  type(REAL_ARRAY) :: TAX
                  type(REAL_ARRAY) :: TSF
                  type(REAL_ARRAY) :: TPF
                  type(INT_ARRAY) :: NDJ4
            END TYPE LATCOM

            INTERFACE
                  FUNCTION wrap_latcom(NWLAT_local, DTM4_local, TST4_local, TTS4_local, TTB4_local, FROST4_local, AFRO4_local, TTA4_local, TTX4_local, TMN4_local, TIN_local, TAX_local, TSF_local, TPF_local, NDJ4_local) RESULT (C_LATCOM) BIND(C)
                        USE iso_c_binding
                        import :: LATCOM
                        TYPE(LATCOM) :: C_LATCOM
                        INTEGER(C_INT), INTENT(in), VALUE :: NWLAT_local
                        TYPE(C_PTR), INTENT(in), VALUE :: DTM4_local
                        TYPE(C_PTR), INTENT(in), VALUE :: TST4_local
                        TYPE(C_PTR), INTENT(in), VALUE :: TTS4_local
                        TYPE(C_PTR), INTENT(in), VALUE :: TTB4_local
                        TYPE(C_PTR), INTENT(in), VALUE :: FROST4_local
                        TYPE(C_PTR), INTENT(in), VALUE :: AFRO4_local
                        TYPE(C_PTR), INTENT(in), VALUE :: TTA4_local
                        TYPE(C_PTR), INTENT(in), VALUE :: TTX4_local
                        TYPE(C_PTR), INTENT(in), VALUE :: TMN4_local
                        TYPE(C_PTR), INTENT(in), VALUE :: TIN_local
                        TYPE(C_PTR), INTENT(in), VALUE :: TAX_local
                        TYPE(C_PTR), INTENT(in), VALUE :: TSF_local
                        TYPE(C_PTR), INTENT(in), VALUE :: TPF_local
                        TYPE(C_PTR), INTENT(in), VALUE :: NDJ4_local
                  END FUNCTION wrap_latcom
            END INTERFACE

            CONTAINS
                  
      END MODULE latcm8m_struct

File: src/module/c/latc8m_struct.c
# include <stdlib.h>

# include "latc8m_struct.h"

latc8m_struct wrap_latc8m_struct(int param_nwlat, f_real_array dtm4, f_real_array tst4, f_real_array tts4, f_real_array ttb4, f_real_array frost4, f_real_array afro4, f_real_array tta4, f_real_array ttx4, nd_real_array tmn4, nd_real_array tin, nd_real_array tax, nd_real_array tsf, nd_real_array tpf, f_int_array ndj4) {
  latc8m_struct wrapped_struct;
  wrapped_struct.param_nwlat = param_nwlat;
  wrapped_struct.dtm4 = dtm4;
  wrapped_struct.tst4 = tst4;
  wrapped_struct.tts4 = tts4;
  wrapped_struct.ttb4 = ttb4;
  wrapped_struct.frost4 = frost4;
  wrapped_struct.afro4 = afro4;
  wrapped_struct.tta4 = tta4;
  wrapped_struct.ttx4 = ttx4;
  wrapped_struct.tmn4 = tmn4;
  wrapped_struct.tin = tin;
  wrapped_struct.tax = tax;
  wrapped_struct.tsf = tsf;
  wrapped_struct.tpf = tpf;
  wrapped_struct.ndj4 = ndj4;
  return wrapped_struct;
}